### ISSUE TITLE
Vulnerability roundup 47 (release-18.09)

### ISSUE THREAD

**Issue by ckauhaus** [NixOS/nixpkgs#47122]
Scanned nixos/release-combined.nix @ ef450ef. Filtered out previously reported CVEs. May contain false positives.

### binutils-2.30 ([search](https://search.nix.gsc.io/?q=binutils&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=binutils+in%3Apath&type=Code))
 - [ ] [CVE-2018-10372](https://nvd.nist.gov/vuln/detail/CVE-2018-10372)
 - [ ] [CVE-2018-10373](https://nvd.nist.gov/vuln/detail/CVE-2018-10373)
 - [ ] [CVE-2018-10534](https://nvd.nist.gov/vuln/detail/CVE-2018-10534)
 - [ ] [CVE-2018-10535](https://nvd.nist.gov/vuln/detail/CVE-2018-10535)
 - [ ] [CVE-2018-12641](https://nvd.nist.gov/vuln/detail/CVE-2018-12641)
 - [ ] [CVE-2018-12697](https://nvd.nist.gov/vuln/detail/CVE-2018-12697)
 - [ ] [CVE-2018-12698](https://nvd.nist.gov/vuln/detail/CVE-2018-12698)
 - [ ] [CVE-2018-12699](https://nvd.nist.gov/vuln/detail/CVE-2018-12699)
 - [ ] [CVE-2018-12700](https://nvd.nist.gov/vuln/detail/CVE-2018-12700)
 - [ ] [CVE-2018-12934](https://nvd.nist.gov/vuln/detail/CVE-2018-12934)
 - [ ] [CVE-2018-13033](https://nvd.nist.gov/vuln/detail/CVE-2018-13033)
 - [ ] [CVE-2018-6543](https://nvd.nist.gov/vuln/detail/CVE-2018-6543)
 - [ ] [CVE-2018-6759](https://nvd.nist.gov/vuln/detail/CVE-2018-6759)
 - [ ] [CVE-2018-6872](https://nvd.nist.gov/vuln/detail/CVE-2018-6872)

### coreutils-8.29 ([search](https://search.nix.gsc.io/?q=coreutils&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=coreutils+in%3Apath&type=Code))
 - [x] [CVE-2017-18018](https://nvd.nist.gov/vuln/detail/CVE-2017-18018)

### exempi-2.4.5 ([search](https://search.nix.gsc.io/?q=exempi&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=exempi+in%3Apath&type=Code))
 - [x] [CVE-2018-12648](https://nvd.nist.gov/vuln/detail/CVE-2018-12648)

### ffmpeg-3.4.4 ([search](https://search.nix.gsc.io/?q=ffmpeg&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=ffmpeg+in%3Apath&type=Code))
 - [x] [CVE-2018-14394](https://nvd.nist.gov/vuln/detail/CVE-2018-14394)
 - [x] [CVE-2018-14395](https://nvd.nist.gov/vuln/detail/CVE-2018-14395)

### glibc-2.27 ([search](https://search.nix.gsc.io/?q=glibc&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=glibc+in%3Apath&type=Code))
 - [ ] [CVE-2018-11236](https://nvd.nist.gov/vuln/detail/CVE-2018-11236)
 - [ ] [CVE-2018-11237](https://nvd.nist.gov/vuln/detail/CVE-2018-11237)

### graphviz-2.40.1 ([search](https://search.nix.gsc.io/?q=graphviz&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=graphviz+in%3Apath&type=Code))
 - [ ] [CVE-2018-10196](https://nvd.nist.gov/vuln/detail/CVE-2018-10196)

### gstreamer-0.10.36 ([search](https://search.nix.gsc.io/?q=gstreamer&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=gstreamer+in%3Apath&type=Code))
 - [x] [CVE-2016-9447](https://nvd.nist.gov/vuln/detail/CVE-2016-9447)

### jasper-2.0.14 ([search](https://search.nix.gsc.io/?q=jasper&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=jasper+in%3Apath&type=Code))
 - [ ] [CVE-2018-9252](https://nvd.nist.gov/vuln/detail/CVE-2018-9252)

### jquery-ui-1.11.4 ([search](https://search.nix.gsc.io/?q=jquery-ui&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=jquery-ui+in%3Apath&type=Code))
 - [ ] [CVE-2016-7103](https://nvd.nist.gov/vuln/detail/CVE-2016-7103)

### libarchive-3.3.2 ([search](https://search.nix.gsc.io/?q=libarchive&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=libarchive+in%3Apath&type=Code))
 - [x] [CVE-2017-14501](https://nvd.nist.gov/vuln/detail/CVE-2017-14501)
 - [x] [CVE-2017-14503](https://nvd.nist.gov/vuln/detail/CVE-2017-14503)

### libcroco-0.6.12 ([search](https://search.nix.gsc.io/?q=libcroco&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=libcroco+in%3Apath&type=Code))
 - [ ] [CVE-2017-8834](https://nvd.nist.gov/vuln/detail/CVE-2017-8834)
 - [ ] [CVE-2017-8871](https://nvd.nist.gov/vuln/detail/CVE-2017-8871)

### libid3tag-0.15.1b ([search](https://search.nix.gsc.io/?q=libid3tag&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=libid3tag+in%3Apath&type=Code))
 - [x] [CVE-2017-11550](https://nvd.nist.gov/vuln/detail/CVE-2017-11550)
 - [x] [CVE-2017-11551](https://nvd.nist.gov/vuln/detail/CVE-2017-11551)

### libmad-0.15.1b ([search](https://search.nix.gsc.io/?q=libmad&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=libmad+in%3Apath&type=Code))
 - [x] [CVE-2018-7263](https://nvd.nist.gov/vuln/detail/CVE-2018-7263)

### libsass-3.5.4 ([search](https://search.nix.gsc.io/?q=libsass&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=libsass+in%3Apath&type=Code))
 - [ ] [CVE-2018-11499](https://nvd.nist.gov/vuln/detail/CVE-2018-11499)
 - [ ] [CVE-2018-11693](https://nvd.nist.gov/vuln/detail/CVE-2018-11693)
 - [ ] [CVE-2018-11694](https://nvd.nist.gov/vuln/detail/CVE-2018-11694)
 - [ ] [CVE-2018-11696](https://nvd.nist.gov/vuln/detail/CVE-2018-11696)
 - [ ] [CVE-2018-11697](https://nvd.nist.gov/vuln/detail/CVE-2018-11697)
 - [ ] [CVE-2018-11698](https://nvd.nist.gov/vuln/detail/CVE-2018-11698)

### libsndfile-1.0.28 ([search](https://search.nix.gsc.io/?q=libsndfile&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=libsndfile+in%3Apath&type=Code))
 - [ ] [CVE-2017-14245](https://nvd.nist.gov/vuln/detail/CVE-2017-14245)
 - [ ] [CVE-2017-14246](https://nvd.nist.gov/vuln/detail/CVE-2017-14246)
 - [x] [CVE-2018-13139](https://nvd.nist.gov/vuln/detail/CVE-2018-13139) ad9e8ea
 - [x] [CVE-2018-13419](https://nvd.nist.gov/vuln/detail/CVE-2018-13419)

### libtiff-4.0.9 ([search](https://search.nix.gsc.io/?q=libtiff&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=libtiff+in%3Apath&type=Code))
 - [ ] [CVE-2017-17095](https://nvd.nist.gov/vuln/detail/CVE-2017-17095)
 - [ ] [CVE-2018-10126](https://nvd.nist.gov/vuln/detail/CVE-2018-10126)
 - [ ] [CVE-2018-10963](https://nvd.nist.gov/vuln/detail/CVE-2018-10963)
 - [ ] [CVE-2018-12900](https://nvd.nist.gov/vuln/detail/CVE-2018-12900)
 - [ ] [CVE-2018-5784](https://nvd.nist.gov/vuln/detail/CVE-2018-5784)
 - [ ] [CVE-2018-7456](https://nvd.nist.gov/vuln/detail/CVE-2018-7456)

### libvorbis-1.3.6 ([search](https://search.nix.gsc.io/?q=libvorbis&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=libvorbis+in%3Apath&type=Code))
 - [ ] [CVE-2018-10393](https://nvd.nist.gov/vuln/detail/CVE-2018-10393)

### lua-5.1.5 ([search](https://search.nix.gsc.io/?q=lua&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=lua+in%3Apath&type=Code))
 - [x] [CVE-2014-5461](https://nvd.nist.gov/vuln/detail/CVE-2014-5461)

### net-snmp-5.7.3 ([search](https://search.nix.gsc.io/?q=net-snmp&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=net-snmp+in%3Apath&type=Code))
 - [x] [CVE-2014-2285](https://nvd.nist.gov/vuln/detail/CVE-2014-2285)

### openjpeg-2.3.0 ([search](https://search.nix.gsc.io/?q=openjpeg&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=openjpeg+in%3Apath&type=Code))
 - [ ] [CVE-2017-17479](https://nvd.nist.gov/vuln/detail/CVE-2017-17479)
 - [x] [CVE-2017-17480](https://nvd.nist.gov/vuln/detail/CVE-2017-17480)
 - [ ] [CVE-2018-14423](https://nvd.nist.gov/vuln/detail/CVE-2018-14423)
 - [ ] [CVE-2018-5727](https://nvd.nist.gov/vuln/detail/CVE-2018-5727)
 - [ ] [CVE-2018-5785](https://nvd.nist.gov/vuln/detail/CVE-2018-5785)
 - [ ] [CVE-2018-6616](https://nvd.nist.gov/vuln/detail/CVE-2018-6616)
 - [x] [CVE-2018-7648](https://nvd.nist.gov/vuln/detail/CVE-2018-7648)

### patch-2.7.6 ([search](https://search.nix.gsc.io/?q=patch&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=patch+in%3Apath&type=Code))
 - [ ] [CVE-2018-1000156](https://nvd.nist.gov/vuln/detail/CVE-2018-1000156)
 - [ ] [CVE-2018-6952](https://nvd.nist.gov/vuln/detail/CVE-2018-6952)

### procps-3.3.15 ([search](https://search.nix.gsc.io/?q=procps&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=procps+in%3Apath&type=Code))
 - [x] [CVE-2018-1121](https://nvd.nist.gov/vuln/detail/CVE-2018-1121)

### rsync-3.1.3 ([search](https://search.nix.gsc.io/?q=rsync&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=rsync+in%3Apath&type=Code))
 - [x] [CVE-2017-16548](https://nvd.nist.gov/vuln/detail/CVE-2017-16548)

### sddm-0.17.0 ([search](https://search.nix.gsc.io/?q=sddm&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=sddm+in%3Apath&type=Code))
 - [ ] [CVE-2018-14345](https://nvd.nist.gov/vuln/detail/CVE-2018-14345)

### taglib-1.11.1 ([search](https://search.nix.gsc.io/?q=taglib&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=taglib+in%3Apath&type=Code))
 - [ ] [CVE-2018-11439](https://nvd.nist.gov/vuln/detail/CVE-2018-11439)

### wildmidi-0.4.2 ([search](https://search.nix.gsc.io/?q=wildmidi&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=wildmidi+in%3Apath&type=Code))
 - [ ] [CVE-2017-11661](https://nvd.nist.gov/vuln/detail/CVE-2017-11661)
 - [ ] [CVE-2017-11662](https://nvd.nist.gov/vuln/detail/CVE-2017-11662)
 - [ ] [CVE-2017-11663](https://nvd.nist.gov/vuln/detail/CVE-2017-11663)
 - [ ] [CVE-2017-11664](https://nvd.nist.gov/vuln/detail/CVE-2017-11664)

### wpa_supplicant-2.6 ([search](https://search.nix.gsc.io/?q=wpa_supplicant&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=wpa_supplicant+in%3Apath&type=Code))
 - [x] [CVE-2017-13077](https://nvd.nist.gov/vuln/detail/CVE-2017-13077)
 - [x] [CVE-2017-13078](https://nvd.nist.gov/vuln/detail/CVE-2017-13078)
 - [x] [CVE-2017-13079](https://nvd.nist.gov/vuln/detail/CVE-2017-13079)
 - [x] [CVE-2017-13080](https://nvd.nist.gov/vuln/detail/CVE-2017-13080)
 - [x] [CVE-2017-13081](https://nvd.nist.gov/vuln/detail/CVE-2017-13081)
 - [x] [CVE-2017-13082](https://nvd.nist.gov/vuln/detail/CVE-2017-13082)
 - [x] [CVE-2017-13084](https://nvd.nist.gov/vuln/detail/CVE-2017-13084)
 - [x] [CVE-2017-13086](https://nvd.nist.gov/vuln/detail/CVE-2017-13086)
 - [x] [CVE-2017-13087](https://nvd.nist.gov/vuln/detail/CVE-2017-13087)
 - [x] [CVE-2017-13088](https://nvd.nist.gov/vuln/detail/CVE-2017-13088)

### zip-3.0 ([search](https://search.nix.gsc.io/?q=zip&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=zip+in%3Apath&type=Code))
 - [ ] [CVE-2018-13410](https://nvd.nist.gov/vuln/detail/CVE-2018-13410)

Cc: @joepie91, @phanimahesh, @the-kenny, @7c6f434c, @k0001, @peterhoeg, @nh2, @LnL7, @grahamc, @adisbladis, @fpletz, @vcunat

Contact @ckauhaus for any questions.

**Comment by ckauhaus** [comment]
master is #47121

**Comment by vcunat** [comment]
`coreutils`: @ckauhaus can you whitelist this CVE?  See https://github.com/NixOS/nixpkgs/issues/38993#issuecomment-386791318

**Comment by ckauhaus** [comment]
@vcunat each CVE is reoprted only once per branch anyway

**Comment by periklis** [comment]
@ckauhaus exempi was patched in master by @Ma27 and i see it is merged in release-18.09 branch ([See L12](https://github.com/NixOS/nixpkgs/blob/release-18.09/pkgs/development/libraries/exempi/default.nix#L12))

**Comment by periklis** [comment]
According to RH-Errata (https://access.redhat.com/errata/RHSA-2016:2974) the gstreamer issue affects only the package `gstreamer-plugins-bad` and the affected plugin was removed in `0.10.19`. We have actually `1.14.2` in `release-18.09` and `master`. I've checked the source files and the NSF plugin is not listed anymore.

**Comment by periklis** [comment]
For jquery-ui there is an new release `1.12.1` which includes the [patch](https://github.com/jquery/jquery-ui/pull/1622) : 
https://github.com/jquery/jquery-ui/compare/1-11-stable...1.12

How important is `nixos/lib/testing`?

**Comment by periklis** [comment]
For `libid3tag` a patch is available in [bugzilla](https://bug162647.bugzilla-attachments.gnome.org/attachment.cgi?id=35604). According to [debian](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=870333#10) both CVEs point to the same issue. Thus the patch should solve both.

**Comment by ckauhaus** [comment]
@periklis great job, thanks :)

One request: please tick off the pkgs that you have been reviewed in the list at the top.

**Comment by ckauhaus** [comment]
> For jquery-ui there is an new release `1.12.1` which includes the [patch](https://github.com/jquery/jquery-ui/pull/1622) :
> [jquery/jquery-ui@1-11-stable...1.12](https://github.com/jquery/jquery-ui/compare/1-11-stable...1.12)
> How important is `nixos/lib/testing?

Didn't get it - do you mean that the vuln is exploitable via a nixos test?

**Comment by ckauhaus** [comment]
@periklis Would it be feasible to provide PRs for both libarchive and libid3tag?

**Comment by periklis** [comment]
@ckauhaus I will provide PRs for both libarchive and libid3tag in the next days.

As for jquery, sinse is used only for nixos-testing, i am not sure if we should patch by updating. wdyt?

**Comment by periklis** [comment]
@ckauhaus unfortunately i cannot tick the items in the list at the top.

**Comment by fpletz** [comment]
The wpa_supplicant CVEs have already been fixed last year: ea50efcc67cfa6c8331b54ff33ab791dacd52fe4

**Comment by fpletz** [comment]
We're not affected by CVE-2014-2285 for net_snmp because we have the fixed version (fixed in >= 5.7.3_pre3 and we have 5.7.3) and our version of perl is new enough to be unaffected anyway (same for RHEL as in https://bugzilla.redhat.com/show_bug.cgi?id=1072778).

**Comment by fpletz** [comment]
CVE-2018-7263 in libmad is a [duplicate of CVE-2017-11552 for libao](https://security-tracker.debian.org/tracker/CVE-2018-7263). Doesn't seem to be fixed upstream and [doesn't look to be reproducible](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=870608). Not actionable for now.

**Comment by c0bw3b** [comment]
CVE-2017-17480 fixed in b3aff3a0400a609e5c1415dc6e9b945948175454

**Comment by vcunat** [comment]
rsync CVE-2017-16548: see https://github.com/NixOS/nixpkgs/issues/38993#issuecomment-381612943

**Comment by vcunat** [comment]
procps CVE-2018-1121: I'm satisfied by https://bugzilla.redhat.com/show_bug.cgi?id=1575473#c3

**Comment by Ekleog** [comment]
Closing as 18.09 is no longer supported.
